File: src/main/java/year2020/Day01.java
package year2020;

import aoc.IAocTask;

import java.util.HashSet;
import java.util.List;
import java.util.stream.Collectors;

public class Day01 implements IAocTask {
    @Override
    public String getFileName() {
        return "aoc2020/input_01.txt";
    }

    @Override
    public void solvePartOne(List<String> lines) {
        HashSet<Integer> complements = new HashSet<>();
        List<Integer> expenses = lines.stream()
                .map(Integer::parseInt)
                .collect(Collectors.toList());
        for (var expense : expenses) {
            if (complements.contains(expense)) {
                System.out.println(expense * (2020 - expense));
                break;
            }
            complements.add(2020 - expense);
        }
    }

    @Override
    public void solvePartTwo(List<String> lines) {

    }
}

File: src/main/kotlin/year2021/Day01.kt
package year2021

import aoc.IAocTaskKt

class Day01 : IAocTaskKt {

    override fun getFileName() = "aoc2021/input_01.txt"
    override fun solvePartOne(lines: List<String>) {
        val increases = lines.map { Integer.parseInt(it) }
            .windowed(2, 1) { prevCur -> if (prevCur[0] < prevCur[1]) 1 else 0 }
            .sumOf { it }
        println(increases)
    }

    override fun solvePartTwo(lines: List<String>) {
        val increases = lines.map { Integer.parseInt(it) }
            .windowed(4, 1) { if (it.subList(0, 3).sum() < it.subList(1, 4).sum()) 1 else 0 }
            .sumOf { it }
        println(increases)
    }
}